### DISCUSSION TITLE
ModbusSequentialDataBlock

### DISCUSSION THREAD

**Discussion by zwanghi** [pymodbus-dev/pymodbus#1619] [Q&A] [answered]
Hi,

I have got a question about Sequential data blocks:
This is a snippet form the server_updating.py example
`datablock= ModbusSequentialDataBlock(0x00, [17] * 100)`

What is the reason for assigning here an address (0x00)? 
I could also  write my data into any address used in the setValues function anyways:

`context[slave_id].setValues(fc_as_hex, address, values)`

I read in the docs that 0xFF is the largest but what if I want to go higher? the hoding registers go from 40000 to 49999?

**Comment by janiversen** [comment, marked as answer]
Please read the documentation that helps!

0x00 is the start address. Every register contains exactly 16bit, so that is your value limitation.

Do not use setValues() to setup the datastore even though it might work (if so it is an accident). setValues() like getValues() operate on the array you define the object.
